src/binding/axum/mod.rs

```diff
@@ -0,0 +1,169 @@
+//! This module integrates the [cloudevents-sdk](https://docs.rs/cloudevents-sdk) with [Axum web service framework](https://docs.rs/axum/)
+//! to easily send and receive CloudEvents.
+//!
+//! To deserialize an HTTP request as CloudEvent
+//!
+//! To echo events:
+//!
+//! ```
+//! use axum_lib as axum;
+//! use axum::{
+//!     handler::{get, post},
+//!     routing::BoxRoute,
+//!     Router,
+//! };
+//! use cloudevents::Event;
+//! use http::StatusCode;
+//!
+//! fn app() -> Router<BoxRoute> {
+//!     Router::new()
+//!         .route("/", get(|| async { "hello from cloudevents server" }))
+//!         .route(
+//!             "/",
+//!             post(|event: Event| async move {
+//!                 println!("received cloudevent {}", &event);
+//!                 (StatusCode::OK, event)
+//!             }),
+//!         )
+//!         .boxed()
+//! }
+//!
+//! ```
+//!
+//! To create event inside request handlers and send them as responses:
+//!
+//! ```
+//! use axum_lib as axum;
+//! use axum::{
+//!     handler::{get, post},
+//!     routing::BoxRoute,
+//!     Router,
+//! };
+//! use cloudevents::{Event, EventBuilder, EventBuilderV10};
+//! use http::StatusCode;
+//! use serde_json::json;
+//!
+//! fn app() -> Router<BoxRoute> {
+//!     Router::new()
+//!         .route("/", get(|| async { "hello from cloudevents server" }))
+//!         .route(
+//!             "/",
+//!             post(|| async move {
+//!                 let event = EventBuilderV10::new()
+//!                     .id("1")
+//!                     .source("url://example_response/")
+//!                     .ty("example.ce")
+//!                     .data(
+//!                         mime::APPLICATION_JSON.to_string(),
+//!                         json!({
+//!                             "name": "John Doe",
+//!                             "age": 43,
+//!                             "phones": [
+//!                                 "+44 1234567",
+//!                                 "+44 2345678"
+//!                             ]
+//!                         }),
+//!                     )
+//!                     .build()
+//!                     .map_err(|e| (StatusCode::INTERNAL_SERVER_ERROR, e.to_string()))?;
+//!
+//!                 Ok::<Event, (StatusCode, String)>(event)
+//!             }),
+//!         )
+//!         .boxed()
+//! }
+//!
+//! ```
+
+pub mod extract;
+pub mod response;
+mod server_response;
+
+#[cfg(test)]
+mod tests {
+
```

Comment:
I guess we can move this to `example-project`

Reply:
Agreed.

@dejanb is it part of this projects code of conduct for you to resolve the conversation once satisfied or should I?
Would you like to see the addition to the example projects folder in this pull request?

Reply:
I agree with moving this to be an example in this PR, if it's not too much trouble. And our CoC is pretty flexible wrt convo resolutions. I'd say feel free to mark it resolved once you've made that change?

Reply:
I am happy to create an example in the example-projects folder.
It is still not useful to have module/binding level integration tests?

Reply:
They are useful, yes, so they can remain here and a minimal example can be copied over, perhaps with only a single test just for completeness. We only build the examples, not test them: https://github.com/cloudevents/sdk-rust/blob/master/.github/workflows/rust_tests.yml#L100

Reply:
@jcrossley3 axum-example now part of the pull request, located in example-projects

Reply:
I made some project wide cargo clippy fixes along the way, hope that is ok.

I would say I am now finished with my changes.